### PR TITLE
Add generators for all mocks

### LINKED ISSUE #605: Generate mocks with go generation tags
Add go generate tags to all generated mocks

e.g. in `pkg/app/mocks/generate.go`

```go
//go:generate mockery -dir .. -output . -name=App
```

### PR DESCRIPTION
**Pull request by bryanl**
Add generators for mocks in the following packages:

pkg/cluster/mocks
pkg/app/mocks
pkg/util/github/mocks
pkg/component/mocks
pkg/registry/mocks
pkg/pkg/mocks

Fixes #605

Signed-off-by: bryanl <bryanliles@gmail.com>

### CODE REVIEW

File: pkg/prototype/rice-box.go [hunk 1]
```diff
@@ -25,34 +10,34 @@ func init() {
 	// define files
 	file2 := &embedded.EmbeddedFile{
 		Filename:    "config-map.jsonnet",
-		FileModTime: time.Unix(1526913293, 0),
+		FileModTime: time.Unix(1526918148, 0),
```
Comment (shomron) [inline pkg/prototype/rice-box.go]: Did these changes sneak in?

File: pkg/prototype/rice-box.go [hunk 2]
```diff
@@ -1,18 +1,3 @@
-// Copyright 2018 The ksonnet authors
```
Comment (shomron) [inline pkg/prototype/rice-box.go]: Keeping the copyright notice on generated files has been somewhat a pain point.

Comment (shomron) [reply pkg/prototype/rice-box.go]: It looks like it's at least addressable with _mockery_ - something to look at later:

```
  -note string
    	comment to insert into prologue of each generated file
```

File: pkg/prototype/rice-box.go (hunk 1 above)
Comment (bryanl) [reply pkg/prototype/rice-box.go]: I ran generate and it updated the mtime. I didn't see it as a problem

**Pull request merged**